tests/e2e/specs/delegate-spec.js

```diff
@@ -0,0 +1,79 @@
+import * as ethers from 'ethers';
+import DelegatePage from '../pages/wallet/delegate-page';
+
+const delegatePage = new DelegatePage();
+
+describe('Delegate', () => {
+	before(() => {});
+	context('Page', () => {
+		it('should be accessible from sidenav', () => {
+			cy.visit('/');
+			cy.findByTestId('sidenav-/escrow').trigger('mouseover');
+			cy.findByTestId('sidenav-submenu-/delegate').click();
+			cy.findByTestId('form', { timeout: 30 * 1e3 }).should('be.visible');
+		});
+		it('should delegate & undelegate powers to an account', () => {
+			const { address: delegateAddress } = ethers.Wallet.createRandom();
+			const shortenedDelegateAddress = `${delegateAddress.slice(0, 5)}...${delegateAddress.slice(
+				-3
+			)}`;
+
+			delegatePage.visit();
+			delegatePage.getFormButton().contains('connect wallet').click();
+			cy.findByAltText('MetaMask').click();
+			delegatePage.acceptMetamaskAccessRequest();
+			delegatePage.getFormButton().contains('enter an address');
+
+			// approve burn
+			delegatePage.getFormInput().type(delegateAddress);
+			// delegatePage.getTable().contains('No addresses to show');
+			cy.findByTestId('action-burn').click();
+			delegatePage.getFormButton().contains('delegate').click();
+			delegatePage.confirmMetamaskTransaction();
+			// delegatePage.getTable().not.contains('No addresses to show');
+			delegatePage.getTable().contains(shortenedDelegateAddress, { timeout: 30 * 1e3 });
+			cy.findByTestId(`checkbox-${delegateAddress}-all`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-mint`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-burn`).should('be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-claim`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-trade`).should('not.be.checked');
```

Comment:
`cy.findByTestId` should be placed inside `delegate-page.js` with other page objects; for example:
```
	getCheckbox(delegateAddress, type) {
		return cy.findByTestId(`checkbox-${delegateAddress}-${type}`);
	}
```

It also applies to all other `cy.findByTestId` in this test. They should be placed as page objects in corresponding files.